sherpa-onnx/jni/offline-recognizer.cc

```diff
@@ -333,11 +333,40 @@ Java_com_k2fsa_sherpa_onnx_OfflineRecognizer_createStream(JNIEnv * /*env*/,
 
 SHERPA_ONNX_EXTERN_C
 JNIEXPORT void JNICALL Java_com_k2fsa_sherpa_onnx_OfflineRecognizer_decode(
-    JNIEnv * /*env*/, jobject /*obj*/, jlong ptr, jlong streamPtr) {
-  auto recognizer = reinterpret_cast<sherpa_onnx::OfflineRecognizer *>(ptr);
-  auto stream = reinterpret_cast<sherpa_onnx::OfflineStream *>(streamPtr);
-
-  recognizer->DecodeStream(stream);
+    JNIEnv *env, jobject /*obj*/, jlong ptr, jlong streamPtr) {
+  try {
+    if (ptr == 0) {  // Check if the pointer is null
+      jclass exClass = env->FindClass("java/lang/NullPointerException");
+      if (exClass != nullptr) {
+        env->ThrowNew(exClass, "OfflineRecognizer_decode: "
+                      "OfflineRecognizer pointer is null.");
```

Comment:
Can you write a function to throw an exception so that a lot of duplicate code is reduced to call that function.

Reply:
Yes, good point, let me do that!

Reply:
Updated.